MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -1565,41 +1591,60 @@ public Person findBestInRole(int role, String skill) {
      * @param noZeroMinute If TRUE, then techs with no time remaining will be excluded from the list.
      * @param firstTechId  The ID of the tech that should appear first in the list (assuming active and satisfies the
      *                     noZeroMinute argument)
+     * @param sorted	       If TRUE, then return the list sorted from worst to best
+     * @param eliteFirst   If TRUE and sorted also TRUE, then return the list sorted from best to worst
      * @return The list of active {@link Person}s who qualify as technicians ({@link Person#isTech()}).
      */
-    public ArrayList<Person> getTechs(boolean noZeroMinute, UUID firstTechId) {
-        ArrayList<Person> techs = new ArrayList<>();
+	public ArrayList<Person> getTechs(boolean noZeroMinute, UUID firstTechId, boolean sorted, boolean eliteFirst) {
+	    ArrayList<Person> techs = new ArrayList<>();
+
+	    // Get the first tech.
+	    Person firstTech = getPerson(firstTechId);
+	    if ((firstTech != null) && firstTech.isTech() && firstTech.isActive() &&
+	            (!noZeroMinute || firstTech.getMinutesLeft() > 0)) {
+	        techs.add(firstTech);
+	    }
 
-        // Get the first tech.
-        Person firstTech = getPerson(firstTechId);
-        if ((firstTech != null) && firstTech.isTech() && firstTech.isActive() &&
-            (!noZeroMinute || firstTech.getMinutesLeft() > 0)) {
-            techs.add(firstTech);
-        }
+	    for (Person p : getPersonnel()) {
+	        if (p.isTech() && p.isActive() && (!p.equals(firstTech)) && (!noZeroMinute || (p.getMinutesLeft() > 0))) {
+	            techs.add(p);
+	        }
+	    }
+	    //also need to loop through and collect engineers on self-crewed vessels
+	    for(Unit u : getUnits()) {
+	        if(u.isSelfCrewed() && !(u.getEntity() instanceof Infantry) && null != u.getEngineer()) {
+	            techs.add(u.getEngineer());
+	        }
+	    }
 
-        for (Person p : getPersonnel()) {
-            if (p.isTech() && p.isActive() && (!p.equals(firstTech)) && (!noZeroMinute || (p.getMinutesLeft() > 0))) {
-                techs.add(p);
-            }
-        }
-        //also need to loop through and collect engineers on self-crewed vessels
-        for(Unit u : getUnits()) {
-        	if(u.isSelfCrewed() && !(u.getEntity() instanceof Infantry) && null != u.getEngineer()) {
-        		techs.add(u.getEngineer());
-        	}
-        }
-        return techs;
-    }
+	    // Return the tech collection sorted worst to best
+	    // Reverse the sort if we've been asked for best to worst
+	    if (sorted) {
+	        Collections.sort(techs, new Comparator<Person>() {
+                @Override
+                public int compare(Person person1, Person person2) {
+                    int retVal = person1.getExperienceLevel(!person1.isTechPrimary() && person1.isTechSecondary())
```

Comment:
I like complex ternary expressions as much as the next guy, but this is pretty unreadable. Can we untangle it?

Reply:
Ternary is pretty standard with comparators, but I have no objection to changing it.